File: extract_uniformat.py
#!/usr/bin/env python3
import sys

try:
    import pandas as pd
    
    # Read the Excel file
    file_path = '/Users/josephagunbiade/Desktop/studio/onyx/Uniformat Building Elements Code.xlsx'
    
    # Try to read all sheets
    xlsx = pd.ExcelFile(file_path)
    print(f'Sheet names: {xlsx.sheet_names}')
    print('\n' + '='*80 + '\n')
    
    # Read each sheet
    for sheet_name in xlsx.sheet_names:
        print(f'Sheet: {sheet_name}')
        print('-'*40)
        df = pd.read_excel(file_path, sheet_name=sheet_name)
        print(df.to_string())
        print('\n' + '='*80 + '\n')
        
except ImportError:
    print("pandas not installed. Trying openpyxl...")
    
    try:
        from openpyxl import load_workbook
        
        wb = load_workbook('/Users/josephagunbiade/Desktop/studio/onyx/Uniformat Building Elements Code.xlsx', read_only=True)
        
        for sheet_name in wb.sheetnames:
            print(f'\nSheet: {sheet_name}')
            print('-'*40)
            sheet = wb[sheet_name]
            
            for row in sheet.iter_rows(values_only=True):
                if any(cell is not None for cell in row):
                    print('\t'.join(str(cell) if cell is not None else '' for cell in row))
                    
    except ImportError:
        print("Neither pandas nor openpyxl is installed.")
        print("Trying to use csv if possible...")
        
        import csv
        import zipfile
        
        # Excel files are zip archives, try to extract data
        try:
            with zipfile.ZipFile('/Users/josephagunbiade/Desktop/studio/onyx/Uniformat Building Elements Code.xlsx', 'r') as z:
                print("Files in Excel archive:")
                for name in z.namelist():
                    print(f"  {name}")
                    if 'sharedStrings.xml' in name or 'sheet1.xml' in name:
                        print(f"\nContent of {name}:")
                        print(z.read(name).decode('utf-8')[:1000])  # First 1000 chars
        except Exception as e:
            print(f"Error reading as zip: {e}")
            
except Exception as e:
    print(f'Error: {e}')

File: extract-uniformat.js
const XLSX = require('xlsx');
const fs = require('fs');
const path = require('path');

try {
  // Read the Excel file
  const filePath = path.join(__dirname, 'Uniformat Building Elements Code.xlsx');
  const workbook = XLSX.readFile(filePath);
  
  console.log('Sheet names:', workbook.SheetNames);
  console.log('\n' + '='.repeat(80) + '\n');
  
  // Process each sheet
  workbook.SheetNames.forEach(sheetName => {
    console.log(`Sheet: ${sheetName}`);
    console.log('-'.repeat(40));
    
    const worksheet = workbook.Sheets[sheetName];
    const jsonData = XLSX.utils.sheet_to_json(worksheet, { header: 1 });
    
    // Print the data
    jsonData.forEach((row, index) => {
      if (row.some(cell => cell !== undefined && cell !== null && cell !== '')) {
        console.log(row.map(cell => cell || '').join('\t'));
      }
    });
    
    console.log('\n' + '='.repeat(80) + '\n');
  });
  
} catch (error) {
  console.error('Error reading Excel file:', error.message);
  
  // Try to read the file as binary and show first few bytes
  try {
    const buffer = fs.readFileSync(path.join(__dirname, 'Uniformat Building Elements Code.xlsx'));
    console.log('\nFile exists, size:', buffer.length, 'bytes');
    console.log('First 16 bytes (hex):', buffer.slice(0, 16).toString('hex'));
  } catch (fsError) {
    console.error('File system error:', fsError.message);
  }
}